cmd/run_test.go

```diff
@@ -81,3 +86,169 @@ func TestRunWithoutConfig(t *testing.T) {
+			tc.content = strings.ReplaceAll(tc.content, "mkdir ", "mkdir "+tempDir+"/")
+			tc.content = strings.ReplaceAll(tc.content, "rm -rf ", "rm -rf "+tempDir+"/")
+
+			// Render the templated TTP first
+			ttp, err := blocks.RenderTemplatedTTP(tc.content, &tc.execConfig)
+			require.NoError(t, err)
+
+			// Handle potential error from RemoveAll within a deferred function
+			defer func() {
+				err := afs.RemoveAll(tempDir) // cleanup temp directory
+				if err != nil {
+					t.Errorf("failed to remove temp directory: %v", err)
+				}
+			}()
+
+			_, err = ttp.RunSteps(tc.execConfig)
+			if tc.wantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			// Determine which directory to check based on the test case content
+			dirName := tempDir + "/testDir"
+			if strings.Contains(tc.content, "testDir2") {
+				dirName = tempDir + "/testDir2"
+			}
+
+			// Check if the directory exists
+			dirExists, err := afero.DirExists(afs, dirName)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedDirExist, dirExists)
+		})
+	}
+}
+
+func TestCleanupDelayFlag(t *testing.T) {
+	afs := afero.NewOsFs()
+	testCases := []struct {
+		name             string
+		content          string
+		execConfig       blocks.TTPExecutionConfig
+		expectedDirExist bool
+		cleanupDelay     time.Duration
+		wantError        bool
+	}{
+		{
+			name: "Test Cleanup Delay Behavior - Directory Creation",
+			content: `
+---
+name: test-cleanup-delay
+steps:
+  - name: step_one
+    inline: mkdir -p TEMP_DIR_PLACEHOLDER/testDirDelay
+    cleanup:
+      inline: rm -rf TEMP_DIR_PLACEHOLDER/testDirDelay
+`,
+			execConfig: blocks.TTPExecutionConfig{
+				CleanupDelaySeconds: 5, // 5 second delay
+			},
+			expectedDirExist: true,
+			cleanupDelay:     5 * time.Second,
+			wantError:        false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a temp directory to work within
+			tempDir, err := afero.TempDir(afs, "", "testCleanupDelay")
+			require.NoError(t, err)
+
+			dirName := filepath.Join(tempDir, "testDirDelay")
+
+			// Update content to work within the temp directory using filepath.Join to ensure platform compatibility
+			tc.content = strings.ReplaceAll(tc.content, "TEMP_DIR_PLACEHOLDER", tempDir)
+
+			// Render the templated TTP first
+			ttp, err := blocks.RenderTemplatedTTP(tc.content, &tc.execConfig)
+			require.NoError(t, err)
+
+			// Handle potential error from RemoveAll within a deferred function
+			defer func() {
+				err := afs.RemoveAll(tempDir) // cleanup temp directory
+				if err != nil {
+					t.Errorf("failed to remove temp directory: %v", err)
+				}
+			}()
+
+			// Capture the start time
+			startTime := time.Now()
+
+			_, err = ttp.RunSteps(tc.execConfig)
+			if tc.wantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			// Wait for the cleanup delay
+			time.Sleep(tc.cleanupDelay)
```

Comment:
This testing approach for verifying delays won't work. The code that you are testing is running in the same thread and executes a synchronous sleep:

```
		if execCtx.Cfg.CleanupDelaySeconds > 0 {
			logging.L().Infof("[*] Sleeping for Requested Cleanup Delay of %v Seconds", execCtx.Cfg.CleanupDelaySeconds)
			time.Sleep(time.Duration(execCtx.Cfg.CleanupDelaySeconds) * time.Second)
		}
```
So it's already waiting the full duration no matter what is in your test. The `time.Sleep` on this line doesn't do anything - you can delete it and the test will still pass. 

You really just want to measure the time difference before and after RunSteps and ensure it's greater than 5 seconds. However, that's going to be a quite annoying large lag to introduce to the  core unit testing process, so I'd recommend just punting this to a later integration test (that runs on PRs but not as part of the "fast loop" developer flow) later leaving the PR with the bugfix and the other test. 

Let me know what you want to do